test/e2e/framework/node/resource.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	nodectlr "k8s.io/kubernetes/pkg/controller/nodelifecycle"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+const (
+	// Poll is how often to Poll pods, nodes and claims.
+	Poll = 2 * time.Second
+
+	// PollShortTimeout is the short timeout value in polling.
+	PollShortTimeout = 1 * time.Minute
+
+	// SingleCallTimeout is how long to try single API calls (like 'get' or 'list'). Used to prevent
+	// transient failures from failing tests.
+	// TODO: client should not apply this timeout to Watch calls. Increased from 30s until that is fixed.
+	SingleCallTimeout = 5 * time.Minute
+
+	// ssh port
+	sshPort = "22"
+
+	// timeout for proxy requests.
+	proxyTimeout = 2 * time.Minute
+)
+
+// FirstNodeAddresses returns the first address of the given type of each node.
+func FirstNodeAddresses(nodelist *v1.NodeList, addrType v1.NodeAddressType) []string {
+	hosts := []string{}
```

Comment:
The original method named `NodeAddresses`. Rename this method to `FirstNodeAddresses` is because `e2enode.NodeAddresses` will cause a golint error (stutters).

Reply:
Does this function need to return `[]string`, wondering why not just `string`? I see it was always like this but might be worth doing a refactor now, or at least leave a TODO for later